src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20241122105739_EES5625_CopyReleaseVersionFieldsToRelease.sql

```diff
@@ -0,0 +1,17 @@
+﻿UPDATE Releases
+SET Releases.PublicationId      = ReleaseVersions.PublicationId,
+    Releases.Slug               = ReleaseVersions.Slug,
+    Releases.Year               = TRY_CAST(ReleaseVersions.ReleaseName AS INT),
+    Releases.TimePeriodCoverage = ReleaseVersions.TimePeriodCoverage,
+    Releases.Updated            = SYSUTCDATETIME()
+FROM Releases
+         INNER JOIN
+     ReleaseVersions
+     ON Releases.Id = ReleaseVersions.ReleaseId
+         INNER JOIN
+     (SELECT RV2.ReleaseId, MAX(RV2.Version) AS Version
+      FROM ReleaseVersions RV2
+      WHERE RV2.Published IS NOT NULL
+         OR (RV2.Published IS NULL AND RV2.Version = 0)
+      GROUP BY ReleaseId) LatestVersion
+     ON ReleaseVersions.ReleaseId = LatestVersion.ReleaseId AND ReleaseVersions.Version = LatestVersion.Version
```

Comment:
I don't think it can be, because the inner table `LatestVersion` with the GROUP BY in both versions only has two columns,  `ReleaseId` which is a release id, and `Version` which is the max published version number or 0 for that release id).

If we switch to using just that inner table and remove the outer `ReleaseVersions` table, we have no values to set the release fields from.

Those values come from joining `ReleaseVersions` with `LatestVersion` on the same release id and version number.

Reply:
You're right! I forgot to consider the fact that you can only select the columns that appear in an aggregate function or the GROUP BY clause...
I wasn't thinking about the columns you'd need at the time, only if the query selected the right rows.

Let's leave it as it is :)